cdxev/build_public_bom.py

```diff
@@ -39,9 +39,36 @@ def remove_internal_information_from_properties(component: dict) -> None:
         component["properties"] = new_properties
 
 
+def clear_component(component: dict[str, Any]) -> dict:
```

Comment:
Thank you for your previous change. What I hadn't noticed before (my bad!) is that the function probably shoudn't return anything. 
You're taking one `component` as a parameter and editing it in place. There is no reason to return the dict again and from a design-perspective it is misleading. It implies to callers that a new, modified dict is returned.
Compare, for instance, the [documentation of the `list.sort()` method](https://docs.python.org/3.11/library/stdtypes.html#list.sort)
> To remind users that it operates by side effect, it does not return the sorted sequence (use [sorted()](https://docs.python.org/3.11/library/functions.html#sorted) to explicitly request a new sorted list instance